src/test/java/org/mockitousage/constructor/CreatingMocksWithConstructorTest.java

```diff
@@ -70,7 +116,25 @@ public void exception_message_when_constructor_not_found() {
             fail();
         } catch (MockitoException e) {
             assertThat(e).hasMessage("Unable to create mock instance of type 'HasConstructor'");
-            assertThat(e.getCause()).hasMessageContaining("0-arg constructor");
+            assertThat(e.getCause()).hasMessageContaining("Please ensure that the target class has a 0-arg constructor.");
+        }
+    }
+
+    static class MultipleConstructors {
+        MultipleConstructors(Integer x) {}
+        MultipleConstructors(Number x) {}
+    }
+
+    @Test
+    public void exception_message_when_multiple_constructors_found() {
+        try {
+            //when
+            MultipleConstructors mock = mock(MultipleConstructors.class, withSettings().useConstructor(123).defaultAnswer(CALLS_REAL_METHODS));
```

Comment:
123 is integer so we should be able to find a matching constructor here, no?

1. I'd like to see the test coverage for those scenarios:

```java
class A {}
class B extends A {}
class C extends A {}

//given constructors:
SomeClass(A a) {}
SomeClass(B b) {}
SomeClass(C c) {}

//this should work:
SomeClass(a);
SomeClass(b);
SomeClass(c);
```

2. Also, I'd like to see coverage for truly ambiguous constructors:

```java
class A {}
class B extends A {}
class C extends B {}

//given constructors
SomeClass(A a) {}
SomeClass(B b) {}

//this is ambiguous and should fail
SomeClass(c)

//not ambiguous and should work:
SomeClass(a)
SomeClass(b)
```